oam/install_scripts/mcs-workernode.service.in

```diff
@@ -1,15 +1,12 @@
 [Unit]
 Description=mcs-workernode
-PartOf=mariadb-columnstore.service
 After=mcs-loadbrm.service
 
 [Service]
 Type=forking
 ExecStart=@ENGINE_BINDIR@/workernode DBRM_Worker1
 Restart=on-failure
-ExecStop=-@ENGINE_BINDIR@/save_brm
+ExecStop=/usr/bin/env bash -c "kill -15 $MAINPID"
+ExecStopPost=-@ENGINE_BINDIR@/save_brm
 ExecStopPost=/usr/bin/env bash -c "clearShm > /dev/null 2>&1"
```

Comment:
Is there a problem with having two ExecStopPost lines?  Maybe need to join them with a &&?  Actually, regardless, probably a good idea to join them anyway.  If save_brm fails, we wouldn't want to run clearShm.

Reply:
To answer your question let's consider a case where save_brm return with 1. This means MCS might left scraps of shmem segments behind.
There are two approaches we can do: forcely clear shmem to allow MCS to start and work properly next time. This is what is achived by these two ExecStopPost. Users might loose the latest meta but at least they have meta backups that can be used in this case.
What is you suggesting is to exit w/o clearShm. When MCS starts next time one'll have multiple shmem access artifacts like you faced recently.
The consensus is to prevent MCS from starting if the original failure happened. We don't have time for this now but can spend time on this before we deliver the multi-node ent release.